core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -571,54 +601,58 @@ boolean putIfAbsentComputeIfPresent(K key, V value, Consumer<OakWBuffer> compute
             /* If {@code lookUp == null} i.e., there is no entry with that key in Oak, the previous version
             associated with the entry is INVALID_VERSION.
             Otherwise, there is an entry with the given key and its value is deleted (or at least in the process of
-            being deleted). After calling {@code finalizeDeletion(Chunk<K, V>, LookUp)}, the version of this entry
+            being deleted). After calling {@code deleteValueFinish(Chunk<K, V>, LookUp)}, the version of this entry
             should be negative.
             Concurrent insertions, however, may cause the version to become valid again.
             Thus, to ensure that the operation behaves as if it itself unlinked the deleted value, we take the minus
             the absolute value of the version written in lookUp (this way, if the version is already negative, it
             remains negative).
              */
-            int oldVersion = lookUp == null ? INVALID_VERSION : -Math.abs(lookUp.version);
 
             // we come here when no key was found, which can be in 3 cases:
             // 1. no entry in the linked list at all
             // 2. entry in the linked list, but the value reference is INVALID_VALUE_REFERENCE
             // 3. entry in the linked list, the value referenced is marked as deleted
             int ei;
```

Comment:
`int ei` is never used in a global context so it shouldn't be here.

This is repetitive because `put()`, `putIfAbsent()` and `putIfAbsentComputeIfPresent()` are almost identical.
You might want to consider merging them.

Reply:
ei - fixed
the similarity of those functions are long time known, however the small differences make it harder to merge. Not in this PR for sure